### PR TITLE
Add support for "No Protocol Fee" Liquidity Bootstrapping Pools

### PR DESCRIPTION
**Pull request by nlordell**
This PR is a follow up to #1488

Here we add support for the `NoProtocolFeeLiquidityBootstrappingPoolFactory`, which is (as the name suggests) a version of `LiquidityBootstrappingPoolFactory` without protocol fees.

It shares all indexing logic with `LiquidityBootstrappingPoolFactory`, so this PR is mostly just the boiler plate to add the new factory deployment information, as well as factory type for configuring Balancer liquidity to index this pool on startup.

### Test Plan

No new logic. Expanded existing unit test to check deployment information is set in generated contract code.

Manual test results:
```
$ cargo test -p shared -- balancer_fetched_pools_match_subgraph --ignored --nocapture
...
2021-12-22T15:14:34.155Z  WARN shared::sources::balancer_v2::pool_fetching::tests: unknown_pools=[]
test sources::balancer_v2::pool_fetching::tests::balancer_fetched_pools_match_subgraph ... ok
...
```

### CODE REVIEW

File: crates/contracts/src/lib.rs [hunk 1]
```diff
@@ -112,6 +116,8 @@ mod tests {
         }
         for network in &[1, 4] {
             assert_has_deployment_address!(BalancerV2Vault for *network);
+            assert_has_deployment_address!(BalancerV2LiquidityBootstrappingPoolFactory for *network);
```
Comment (nlordell) [inline crates/contracts/src/lib.rs:119]: unrelated change that I snuck in.

File: crates/shared/src/sources/balancer_v2/pools/liquidity_bootstrapping.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-//! Module implementing two-token weighted pool specific indexing logic.
+//! Module implementing liquidity bootstrapping pool specific indexing logic.
```
Comment (nlordell) [inline crates/shared/src/sources/balancer_v2/pools/liquidity_bootstrapping.rs:1]: Trivial unrelated change that I snuck in.

Comment (bh2smith) [reply crates/shared/src/sources/balancer_v2/pools/liquidity_bootstrapping.rs:1]: This is part of the previous PR and I also a suggestion comment to fix it there. Just FYI

**Review by bh2smith — APPROVED** [review]
Looks good. I see that these no protocol fee pools are being treated as a completely different pool type. Is there any reason we didn't go with having the same pool type with an additional parameter or are they really all that different?

File: crates/shared/src/sources/balancer_v2/pools/liquidity_bootstrapping.rs (hunk 2 above)
Comment (nlordell) [reply crates/shared/src/sources/balancer_v2/pools/liquidity_bootstrapping.rs:1]: Merged your suggestion :smil

**Comment by nlordell** [comment]
> I see that these no protocol fee pools are being treated as a completely different pool type.

They are the same "pool type" in that they have the same `PoolInfo` and `PoolState` types and produce `WeightedPools`. They also reuse 100% of the logic from the existing LBP implementation. However, they need to be indexed separately because they have a separate factory address. Also, since they are a separate factory, we also want the ability to enable/disable them independently (hence the additional `BalancerFactoryKind` variant).

**Review by fleupold — APPROVED** [review]
```
unknown_pools=[]
```

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2FMdqE46HziuFJTlIwjw%2Fgiphy.gif&token=yay" data-gitmeme-token="yay"></a>
